### PR TITLE
Allow ancient ones to tweak the world upon choosing them

### PR DESCRIPTION
**Pull request by petercombs**
This allows Squidface to reduce the maximum sanity/stamina of characters from the very beginning of the game, rather than letting them effectively have a bonus san/stam that they can't refill.

### CODE REVIEW

**Comment by petercombs** [comment]
I wasn't sure how to write a test, but running the server this has the expected behavior: choosing a character, then Squidface reduces their stats, then changing the choice to another AO restores it to the max.

**Comment by fingoldfish** [comment]
I think this was already fixed in https://github.com/gummyboars/lunboks/commit/43a02beaae2663dd1d03879bec0e25a7a35073b7
Though if we need setup for other ancient ones, we can add it.

**Comment by petercombs** [comment]
I think some mechanism like this will also be useful for adding Masks.

File: eldritch/eldritch.py [hunk 1]
```diff
@@ -772,6 +772,7 @@ def handle_ancient(self, ancient):
     if ancient not in self.all_ancients:
       raise InvalidMove(f"Unknown ancient one {ancient}")
     self.ancient_one = self.all_ancients[ancient]
+    self.ancient_one.setup(self)
```
Comment (fingoldfish) [inline eldritch/eldritch.py:775]: Can I ask you to move this from handle_ancient to handle_start? If we want to do something like adding masks, we don't want to have to remove the masks when the player changes their minds about which ancient one to fight.

**Review by fingoldfish — CHANGES_REQUESTED** [review]

**Review by fingoldfish — APPROVED** [review]

**Pull request merged**